btree/btree.c

```diff
@@ -1421,22 +1690,32 @@ static void ff_find(struct slot *slot, const struct m0_btree_key *key)
 	M0_PRE(key->k_data.ov_vec.v_count[0] == h->ff_ksize);
 	M0_PRE(key->k_data.ov_vec.v_nr == 1);
 
-	while (i + 1 < j) {
+	if (h->ff_used == 0 ||
+	    (memcmp(ff_key(slot->s_node, 0),
+		    key->k_data.ov_buf[0], h->ff_ksize) >= 0)) {
+		slot->s_idx = 0;
+		return;
+	} else if (memcmp(ff_key(slot->s_node, (h->ff_used - 1)),
+			  key->k_data.ov_buf[0], h->ff_ksize) < 0) {
+		slot->s_idx = h->ff_used;
+		return;
+	}
```

Comment:
Hmm, why? This just introduced 2 additional memcmp() in hot path.

Reply:
Hmm.. which cases the old code does not handle? If used is 0, both versions return 0. If the target key is less than the leftmost key, both versions return 0.